dashboard/test/lib/services/partial_registration/user_builder_test.rb

```diff
@@ -0,0 +1,280 @@
+    assert_equal EmailPreference::ACCOUNT_SIGN_UP, @user.email_preference_source
+    assert_equal '0', @user.email_preference_form_kind
+    assert_equal 'yes', @user.email_preference_opt_in
+    assert EmailPreference.where(email: @user.email).last.opt_in
+  end
+
+  test 'builds teacher with nces school in our database' do
+    fake_school = create :school
+    fake_school_info = create :school_info, school_id: fake_school.id
+    setup_partial_user({user_type: 'teacher', school_info_attributes:
+      {
+        schoolId: fake_school.id,
+        schoolName: fake_school.name,
+        schoolType: fake_school.school_type,
+        schoolZip: fake_school.zip,
+        schoolState: fake_school.state,
+        country: fake_school_info.country,
+        fullAddress: fake_school_info.full_address
+      }}
+    )
+
+    assert_creates(User) do
+      @user = Services::PartialRegistration::UserBuilder.call(request: @request)
+    end
+
+    assert_equal fake_school_info.id, @user.school_info_id
+    assert UserSchoolInfo.where(school_info_id: fake_school_info.id, user_id: @user.id)
+  end
+
+  test 'builds teacher that adds their own US school' do
+    school_name = 'Fake US School Name'
+    school_zip = '11111'
+    school_country = 'US'
+    setup_partial_user({user_type: 'teacher', school_info_attributes:
+      {
+        schoolId: SharedConstants::NON_SCHOOL_OPTIONS.CLICK_TO_ADD,
+        schoolName: school_name,
+        schoolZip: school_zip,
+        country: school_country,
+      }}
+    )
+
+    assert_creates(User) do
+      @user = Services::PartialRegistration::UserBuilder.call(request: @request)
+    end
+
+    school_info = SchoolInfo.last
+    assert_nil school_info.school_id
+    assert_equal school_name, school_info.school_name
+    assert_equal school_zip, school_info.zip.to_s
+    assert_equal school_country, school_info.country
+    assert UserSchoolInfo.where(school_info_id: school_info.id, user_id: @user.id)
+  end
+
+  test 'builds teacher that adds their own non-US school' do
+    school_name = 'Fake Mexico School Name'
+    school_country = 'MX'
+    setup_partial_user({user_type: 'teacher', school_info_attributes:
+      {
+        schoolId: SharedConstants::NON_SCHOOL_OPTIONS.CLICK_TO_ADD,
+        schoolName: school_name,
+        country: school_country,
+      }}
+    )
+
+    assert_creates(User) do
+      @user = Services::PartialRegistration::UserBuilder.call(request: @request)
+    end
+
+    school_info = SchoolInfo.last
+    assert_nil school_info.school_id
+    assert_equal school_name, school_info.school_name
+    assert_equal school_country, school_info.country
+    assert UserSchoolInfo.where(school_info_id: school_info.id, user_id: @user.id)
+  end
+
+  test 'builds teacher that does not teach in a school setting' do
+    school_country = 'US'
+    setup_partial_user({user_type: 'teacher', school_info_attributes:
+      {
+        schoolId: SharedConstants::NON_SCHOOL_OPTIONS.NO_SCHOOL_SETTING,
```

Comment:
I take full responsibility for the confusion here, but in order to create a no-school setting school info, the `school_type` needs to be `NO_SCHOOL_SETTING` rather than `school_id`

I'm seeing some other divergence wrt school info since that has been a moving target the last couple months... 😬 

I have a [ticket](https://codedotorg.atlassian.net/browse/ACQ-2165) to audit this anyway, so no need for changes at this point I don't think!